app/models/request/sample_compound_aliquot_transfer.rb

```diff
@@ -50,14 +50,36 @@ def aliquots_by_tags_combination
     asset.aliquots.group_by(&:tags_combination)
   end
 
+  # For a group of source aliquots, find or create a compound sample containing the component samples
+  # Assign the compound sample to the target asset
   def transfer_into_compound_sample_aliquot(source_aliquots)
     compound_aliquot = CompoundAliquot.new(request: self, source_aliquots: source_aliquots)
     unless compound_aliquot.valid?
       raise Request::SampleCompoundAliquotTransfer::Error, compound_aliquot.errors.full_messages
     end
 
-    compound_aliquot.create_compound_sample
+    # Check if a compound sample already exists with the source_aliquot samples
+    existing_compound_sample = find_compound_sample_for_component_samples(source_aliquots)
 
+    if existing_compound_sample
+      compound_aliquot.compound_sample = existing_compound_sample
+    else
+      compound_aliquot.create_compound_sample
+    end
     target_asset.aliquots.create(compound_aliquot.aliquot_attributes)
   end
+
+  # Due to previous implementation, there may be multiple compound samples with the provided component samples.
+  # NPG have confirmed we do not need to fix the data where there are multiple compound samples with
+  # the same component samples
+  def find_compound_sample_for_component_samples(source_aliquots)
+    # Get all the component samples
+    component_samples = source_aliquots.map(&:sample)
+
+    # Get all the compound samples of the first component sample
+    compound_samples = component_samples[0].compound_samples.order(:id)
```

Comment:
Rather than ordering by id, then calling reverse, you can simply `order(id: :desc)`
I'd also suggest doing `compound_samples.include(:samples).order(id: :desc)` as it will eager load the samples upfront, otherwise you risk lots of hits to the database.